packages/rich-text/src/plugins/EmbeddedEntityInline/index.tsx

```diff
@@ -89,24 +90,19 @@ async function selectEntityAndInsert(editor, sdk: FieldExtensionSDK) {
     ...newEntitySelectorConfigFromRichTextField(sdk.field, INLINES.EMBEDDED_ENTRY),
     withCreate: true,
   };
+  const selection = editor.selection;
```

Comment:
I wouldn't recommend doing this, I'd rather create a new object to keep the values where they are at the moment of the selection, `editor.selection` can change, therefore `selection` will also be different.

Reply:
`editor.selection` object reference can change, but the selection object props themselves are immutable (read only props).

If you'd want to work with the assumption that the object could be mutated for some reason, then you should use `cloneDeep(editor.selection)` as there's some nesting.